control-plane/agents/src/bin/core/controller/states.rs

```diff
@@ -180,6 +180,14 @@ impl ResourceStates {
         self.replicas.get(id)
     }
 
+    /// Get a snapshot with the given ID.
+    pub(crate) fn snapshot_state(
+        &self,
+        id: &SnapshotId,
+    ) -> Option<&ResourceMutex<ReplicaSnapshotState>> {
+        self.snapshots.get(id)
+    }
+
     /// Clear all state information.
     pub(crate) fn clear_all(&mut self) {
```

Comment:
Does clear_all need to include snapshots?

Reply:
Fixed, good catch!